rumqttd/src/segments/segment.rs

```diff
@@ -135,11 +142,14 @@ mod tests {
         segment.push(Bytes::from_static(b"test9"));
         assert_eq!(segment.len(), 9);
 
-        let mut out: Vec<Bytes> = Vec::new();
-        let _ = segment.readv(0, 2, &mut out).unwrap();
```

Comment:
See if we can add any more unit tests

Reply:
i don't think we can add any more unit test here, `Segments` is just a wrapper around vec with some more metadata. `absolute_offset` stays constant for a particular segment and `size` is tested for by unit tests in `segments/mod.rs`